_posts/2019/2019-05-11-how-to-improve-your-components-with-react-hooks.md

```diff
@@ -0,0 +1,183 @@
+---
+layout: post
+title: How to improve your components with React Hooks
+date: 2019-05-11
+header_image: public/react-hooks.jpg
+header_position: center
+category: coding
+tags: ["javascript", "reactjs"]
+authors: ["Jonathan"]
+about_authors: ["jwieben"]
+---
+
+## How to improve your components with React Hooks
+
+The React world has been going crazy over the new [Hooks API](https://reactjs.org/docs/hooks-overview.html){:target="_blank"} that was released a few months ago. New libraries for Hooks are appearing left and right and everyone seems to be rewriting their apps with them.
+
+So what is it about Hooks that people are so excited about? Well, there are a variety of reasons. In this post I would like to focus on one of the central advantages that Hooks bring to the table: **They allow you to organize your component logic in a simpler way.**
+
+I am going to illustrate this by writing the same component in two ways, with and without Hooks, and later compare them.
+
+#### A typical class component
+
+Let's assume we wanted to write a `UserProfile` component, that displays a users profile picture next to their name. We also want the profile picture to be hidden when the window width is fewer than 1024 pixels (you'd probably use CSS for that, but we are going to do it the JS-way for demonstrational purposes).
```

Comment:
```suggestion
Let's assume we wanted to write a `UserProfile` component, that displays a users profile picture next to their name. We also want the profile picture to be hidden when the window width is less than 1024 pixels (you'd probably use CSS for that, but we are going to do it the JS-way for demonstrational purposes).
```

Reply:
I blindly followed my Grammarly suggestion here... But you're right. :+1: